### PR TITLE
Feature/RR-1352-Win-Breakdown

### PR DESCRIPTION
**Pull request by ince-dbt**
### Description of change

This PR aims to add relevant breakdown values to total expected export, non export and total odi in win.

Below is further detail in relation to the calculation mentioned abovee:

- The value in total expected export value should be the total export value in breakdown
- The value in total expected non export value should be the total business success in breakdown
- The value in total expected odi value should be the total outward direct investment value in breakdown

The total values of total export, non export and total odi  should also get updated whenever the respective values in breakdown change

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/export_win/models.py [hunk 1]
```diff
@@ -42,6 +44,28 @@ def soft_deleted(self, *args, **kwargs):
         )
 
 
+class BaseExportWinTotalCalculation(models.Manager):
```
Comment (elcct) [inline datahub/export_win/models.py:47]: The methods of this class seem to be always used together, they also don't seem to depend on any class members, so they could be collapsed to one function that calculates this at once, like:

```
  def _calculate_totals_for_export_win(win_instance):
        export_type_value = constants.BreakdownType.export.value
        non_export_value = constants.BreakdownType.non_export.value
        odi_value = constants.BreakdownType.odi.value
        return {
            'total_export_value': win_instance.breakdowns.filter(
                type_id=export_type_value.id).aggregate(
                total_export_value=Sum('value')
            )['total_export_value'] or 0,
            'total_non_export_value': win_instance.breakdowns.filter(
                type_id=non_export_value.id).aggregate(
                total_non_export_value=Sum('value')
            )['total_non_export_value'] or 0,
            'total_odi_value': win_instance.breakdowns.filter(
                type_id=odi_value.id).aggregate(
                total_odi_value=Sum('value')
            )['total_odi_value'] or 0,
        }
```

and then used like:

```
    totals = _calculate_totals_for_export_win(self)
    self.total_expected_export_value = totals['total_export_value']
    self.total_expected_non_export_value = totals['total_non_export_value']
    self.total_expected_odi_value = totals['total_odi_value']
```

File: datahub/export_win/models.py [hunk 2]
```diff
@@ -656,3 +688,14 @@ class DeletedWin(Win):
...
     class Meta:
         proxy = True
+
+
+@receiver(post_save, sender=Breakdown)
+def update_total_values(sender, instance, **kwargs):
+    """Save the right total values"""
+    win = instance.win
+    calc_total = BaseExportWinTotalCalculation()
+    win.total_expected_export_value = calc_total.calculate_total_export_value(win)
```
Comment (chopkinsmade) [inline datahub/export_win/models.py:698]: This call would then become `calc_total.update_totals(win)`

File: datahub/export_win/models.py (hunk 1 above)
Comment (ince-dbt) [reply datahub/export_win/models.py:47]: Thanks for taking time to suggest the code refactor.. it is greatly appreciated!... Made changes on this!

File: datahub/export_win/models.py [hunk 3]
```diff
@@ -42,6 +44,28 @@ def soft_deleted(self, *args, **kwargs):
         )
 
 
+class BaseExportWinTotalCalculation(models.Manager):
+    """Base class for Total Export, Non Export and ODI"""
+
+    def calculate_total_export_value(self, win_instance):
```
[the remark this answers is not in the document]
Comment (ince-dbt) [reply datahub/export_win/models.py]: Thanks for taking time to suggest the code refactor.. it is greatly appreciated!... Your is similiar to Pawel. So, have adopted it and update the changes on this!

File: datahub/export_win/models.py (hunk 2 above)
Comment (ince-dbt) [reply datahub/export_win/models.py]: Thanks.. made changes on this..

File: datahub/export_win/models.py (hunk 1 above)
Comment (elcct) [reply datahub/export_win/models.py:47]: Just one more thing - you are still using `class BaseExportWinTotalCalculation(models.Manager)` - there is no real benefit to have this, as the function is really independent of this class and `total_calculation_objects` is not being used. In the example below:
```
    calc_total = BaseExportWinTotalCalculation()._calculate_totals_for_export_win(win)
    win.total_expected_export_value = calc_total['total_export_value']
    win.total_expected_non_export_value = calc_total['total_non_export_value']
    win.total_expected_odi_value = calc_total['total_odi_value']
```
This `BaseExportWinTotalCalculation()` doesn't really do anything apart from instantiating its base class and providing reference to `_calculate_totals_for_export_win` and that can be entirely skipped.

It this was just an ordinary function (not a method of any class) you could simply do:

```
calc_total = _calculate_totals_for_export_win(win)
```

**Review by elcct — APPROVED** [review]

**Pull request merged**